torchgeo/datasets/mmearth.py

```diff
@@ -0,0 +1,505 @@
+        'sentinel2_scl',
+    )
+
+    # See https://github.com/vishalned/MMEarth-train/blob/8d6114e8e3ccb5ca5d98858e742dac24350b64fd/MODALITIES.py#L108C1-L160C2
+    all_modality_bands: ClassVar[dict[str, list[str]]] = {
+        'sentinel2': [
+            'B1',
+            'B2',
+            'B3',
+            'B4',
+            'B5',
+            'B6',
+            'B7',
+            'B8A',
+            'B8',
+            'B9',
+            'B10',
+            'B11',
+            'B12',
+        ],
+        'sentinel2_cloudmask': ['QA60'],
+        'sentinel2_cloudprod': ['MSK_CLDPRB'],
+        'sentinel2_scl': ['SCL'],
+        'sentinel1': [
+            'asc_VV',
+            'asc_VH',
+            'asc_HH',
+            'asc_HV',
+            'desc_VV',
+            'desc_VH',
+            'desc_HH',
+            'desc_HV',
+        ],
+        'aster': ['elevation', 'slope'],
+        'era5': [
+            'prev_month_avg_temp',
+            'prev_month_min_temp',
+            'prev_month_max_temp',
+            'prev_month_total_precip',
+            'curr_month_avg_temp',
+            'curr_month_min_temp',
+            'curr_month_max_temp',
+            'curr_month_total_precip',
+            'year_avg_temp',
+            'year_min_temp',
+            'year_max_temp',
+            'year_total_precip',
+        ],
+        'dynamic_world': ['landcover'],
+        'canopy_height_eth': ['height', 'std'],
+        'lat': ['sin', 'cos'],
+        'lon': ['sin', 'cos'],
+        'biome': ['biome'],
+        'eco_region': ['eco_region'],
+        'month': ['sin_month', 'cos_month'],
+        'esa_worldcover': ['map'],
+    }
+
+    # See https://github.com/vishalned/MMEarth-train/blob/8d6114e8e3ccb5ca5d98858e742dac24350b64fd/MODALITIES.py#L36
+    no_data_vals: ClassVar[dict[str, int | float]] = {
+        'sentinel2': 0,
+        'sentinel2_cloudmask': 65535,
+        'sentinel2_cloudprod': 65535,
+        'sentinel2_scl': 255,
+        'sentinel1': float('-inf'),
+        'aster': float('-inf'),
+        'canopy_height_eth': 255,
+        'dynamic_world': 0,
+        'esa_worldcover': 255,
+        'lat': float('-inf'),
+        'lon': float('-inf'),
+        'month': float('-inf'),
+        'era5': float('inf'),
+        'biome': 255,
+        'eco_region': 65535,
+    }
+
+    norm_modes: ClassVar[list[str]] = ['z-score', 'min-max']
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        ds_version: str = 'MMEarth',
+        modalities: Sequence[str] = all_modalities,
+        modality_bands: dict[str, list[str]] | None = None,
+        split: str = 'train',
+        normalization_mode: str = 'z-score',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+    ) -> None:
+        """Initialize the MM-Earth dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            ds_version: one of "MMEarth", "MMEarth64", or "MMEarth100k"
```

Comment:
How much added complexity was it to support all 3? If it wasn't hard, great! If it's annoying, we could stick to the big version only.